pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -697,18 +698,26 @@ public long getCommitTimeoutMS(String tableName) {
 
   public static int getRealtimeTableFlushSize(TableConfig tableConfig) {
     final Map<String, String> streamConfigs = tableConfig.getIndexingConfig().getStreamConfigs();
-    if (streamConfigs != null && streamConfigs.containsKey(
-        CommonConstants.Helix.DataSource.Realtime.REALTIME_SEGMENT_FLUSH_SIZE)) {
-      final String flushSizeStr =
-          streamConfigs.get(CommonConstants.Helix.DataSource.Realtime.REALTIME_SEGMENT_FLUSH_SIZE);
+    String flushSizeStr;
+    if (streamConfigs == null) {
+      return -1;
+    }
+    if (streamConfigs.containsKey(CommonConstants.Helix.DataSource.Realtime.LLC_REALTIME_SEGMENT_FLUSH_SIZE)) {
+      flushSizeStr = streamConfigs.get(CommonConstants.Helix.DataSource.Realtime.LLC_REALTIME_SEGMENT_FLUSH_SIZE);
       try {
         return Integer.parseInt(flushSizeStr);
-      } catch (Exception e) {
-        LOGGER.warn("Failed to parse flush size of {}", flushSizeStr, e);
-        return -1;
+      } catch (Exception e1) {
+        LOGGER.warn("Failed to parse LLC flush size of {}", flushSizeStr, e1);
+        if (streamConfigs.containsKey(CommonConstants.Helix.DataSource.Realtime.REALTIME_SEGMENT_FLUSH_SIZE)) {
```

Comment:
Not 100% sure about the indent, but is the code `if (llcFlushSize) { if (segmentFlushSize) }`? Those two conditionals should be side by side, not within one another.

Reply:
good catch. i will fix it so that we return -1 only if we can't find any config.